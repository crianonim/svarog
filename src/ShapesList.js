import React from 'react';
import ShapeItem from './ShapeItem';

const ShapesList = ({shapes,selectedShape,setShapes,moveShape}) => {
    return (<>
     <div className="shapes-list">
      {shapes.map( shape => (
        <div key={shape.id} className={"flex-row " +(shape.id===selectedShape?"selected-shape":"")}>
        <button  title="Delete Shape" onClick={()=>{
          setShapes(shapes.filter(el=>el!==shape));
        }}>x</button>
        <ShapeItem  shape={shape} duplicate={(shape)=>{
          const dup=Object.assign({},shape);
          dup.id=Date.now();
          let index=shapes.findIndex(el=>el===shape);
          setShapes([...shapes.slice(0,index+1),dup,...shapes.slice(index+1)]);
        }} selected={selectedShape===shape.id}
        changed={(changedShape)=>{
          shapes[shapes.findIndex((sh)=>sh===changedShape)]=changedShape;
          setShapes(shapes.slice());
        }}
        shapeUp={moveShape(-1)}
        shapeDown={moveShape(1)}
        ></ShapeItem>
        </div>
      ))}
       </div>
    
    </>)
}

export default ShapesList;