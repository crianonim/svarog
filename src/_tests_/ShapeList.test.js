import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import ShapesList from "../components/ShapesList";
it("matches a snapshrot", () => {
  const moveShape = jest.fn();
  const setShapes = jest.fn();
  const setSelectedShape = jest.fn();
  const wrapper = shallow(
    <ShapesList
      selectedShape={1}
      setSelectedShape={setSelectedShape}
      moveShape={moveShape}
      setShapes={setShapes}
      svgAttrs={{ fill: "black" }}
      shapes={[{ id:1,shape: "circle", attributes: { x: 10, y: 20, r: 30 } }]}
    />
  );
  expect(toJson(wrapper)).toMatchSnapshot();
});
