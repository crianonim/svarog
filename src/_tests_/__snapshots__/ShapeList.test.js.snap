// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches a snapshrot 1`] = `
<Fragment>
  <div
    className="shapes-list"
  >
    <div
      className="flex-row selected-shape"
      key="1"
    >
      <button
        className="button is-small is-danger "
        onClick={[Function]}
        title="Delete Shape"
      >
        <FontAwesomeIcon
          border={false}
          className=""
          fixedWidth={false}
          flip={null}
          icon={
            Object {
              "icon": Array [
                448,
                512,
                Array [],
                "f1f8",
                "M432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16zM53.2 467a48 48 0 0 0 47.9 45h245.8a48 48 0 0 0 47.9-45L416 128H32z",
              ],
              "iconName": "trash",
              "prefix": "fas",
            }
          }
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size={null}
          spin={false}
          symbol={false}
          title=""
          transform={null}
        />
      </button>
      <ShapeItem
        duplicate={[Function]}
        selected={true}
        setSelectedShape={[MockFunction]}
        shape={
          Object {
            "attributes": Object {
              "r": 30,
              "x": 10,
              "y": 20,
            },
            "id": 1,
            "shape": "circle",
          }
        }
      />
    </div>
  </div>
</Fragment>
`;
